### PR TITLE
palindrome-products: update tests to v1.2.0

### LINKED ISSUE #626: Update tests for palindrome-products
Tests for `palindrome-products` exercise need to be synced up with [canonical specs](https://github.com/exercism/problem-specifications/blob/master/exercises/palindrome-products/canonical-data.json).

One notable change is - current tests include [`numericArrayMatch`](https://github.com/exercism/javascript/blob/master/exercises/palindrome-products/palindrome-products.spec.js#L46-L59) which can be replaced with a combination of tests like this:

```js
expect(largest.factors).toEqual(expect.arrayContaining([3, 3]));
expect(largest.factors).toEqual(expect.arrayContaining([1, 9]));
expect(largest.factors.length).toEqual(2);
```

Reference from [Jest docs](https://jestjs.io/docs/en/expect#expectarraycontainingarray).

### PR DESCRIPTION
**Pull request by cmccandless**
Resolves #626

### CODE REVIEW

**Comment by SleeplessByte** [comment]
@cmccandless would you be so kind to resolve this so we can get it merged?

**Comment by cmccandless** [comment]
@SleeplessByte resolved!

**Comment by cmccandless** [comment]
Accidental close. Please ignore that.

**Comment by cmccandless** [comment]
Merge/rebase issue has been cleaned up. Once Travis finished, this should be good for squash-merge.

**Comment by cmccandless** [comment]
@SleeplessByte OK, now I'm lost. What of my changes would've affected the Travis build like that? I'm just ran `make test-travis` locally, and it passed just fine, and I didn't commit any changes to package-lock files or anything like that.

**Comment by SleeplessByte** [comment]
@tejasbubane have you seen this weird `Symbol` issue before?

**Comment by tejasbubane** [comment]
This is an issue with Jest and latest node on travis. Fixed by using LTS in https://github.com/exercism/javascript/pull/634. Please merge or rebase this branch on master.

**Comment by cmccandless** [comment]
@tejasbubane rebased, thanks for the info!

File: exercises/palindrome-products/palindrome-products.spec.js [hunk 1]
```diff
@@ -1,69 +1,125 @@
...
+  xtest('error for largest if min is more than max', () => {
+    expect(() => {
+      const palindromes = Palindromes.generate({ maxFactor: 1, minFactor: 2 });
+      palindromes.largest;
+    }).toThrow(new Error('min must be <= max'));
   });
 });
 
-function numericalArraysMatch(a, b) {
+function factorsMatch(a, b) {
```
Comment (tejasbubane) [inline exercises/palindrome-products/palindrome-products.spec.js]: I don't see this function used anywhere? Am I missing something?

Comment (cmccandless) [reply exercises/palindrome-products/palindrome-products.spec.js]: Nope it's not.

I had deleted it, must've crept back in during my rebasing. I will remove it.

**Pull request merged**